unittests/test_hochfrequenz_condition_resolver.py

```diff
@@ -0,0 +1,99 @@
+"""
+Tests the HochfrequenzConditionResolver
+"""
+import asyncio
+import datetime
+
+import pytest  # type:ignore[import]
+from ahbicht.mapping_results import ConditionKeyConditionTextMapping
+
+# https://github.com/pnuckowski/aioresponses/issues/206
+from aioresponses import CallbackResult, aioresponses  # type:ignore[import]
+from maus.edifact import EdifactFormat, EdifactFormatVersion
+
+from ahbichtfunctionsclient import HochfrequenzConditionResolver
+
+pytestmark = pytest.mark.asyncio
+
+
+class TestHochfrequenzConditionResolver:
+    async def test_hochfrequenz_condition_api_success(self):
+        condition_resolver = HochfrequenzConditionResolver(
+            edifact_format_version=EdifactFormatVersion.FV2204,
+            edifact_format=EdifactFormat.UTILMD,
+            api_url="https://test.inv",
+        )
+        with aioresponses() as mocked_server:
+            mocked_server.get(
+                "https://test.inv/FV2204/UTILMD/10P",
+                payload={
+                    "condition_text": "Wenn foo[56] und bar[99] dann FooBar.",
+                    "condition_key": "10P",
+                    "edifact_format": "UTILMD",
+                },
+            )
+            actual = await condition_resolver.get_condition_text("10P")
+            assert actual == ConditionKeyConditionTextMapping(
+                edifact_format=EdifactFormat.UTILMD,
+                condition_key="10P",
+                condition_text="Wenn foo[56] und bar[99] dann FooBar.",
+            )
+
+    async def test_hochfrequenz_condition_api_failure(self):
+        condition_resolver = HochfrequenzConditionResolver(
+            edifact_format_version=EdifactFormatVersion.FV2204,
+            edifact_format=EdifactFormat.UTILMD,
+            api_url="https://test.inv",
+        )
+
+        def simulate_error(url, **kwargs):
+            return CallbackResult(status=400, payload={"it is not": "important what's here, just that you had to wait"})
+
+        with aioresponses() as mocked_server:
+            mocked_server.get(url="https://test.inv/FV2204/UTILMD/001", callback=simulate_error, repeat=5)
```

Comment:
warun hier das repeat? brauchst du glaube ich nicht.
```suggestion
            mocked_server.get(url="https://test.inv/FV2204/UTILMD/001", callback=simulate_error)
```